### ISSUE TITLE
Add remaining CLI functionality for SPL Token Group

### ISSUE THREAD

**Issue by buffalojoec** [solana-labs/solana-program-library#5937]
#### Problem
As of #5911 and #5912, the SPL Token CLI supports adding `GroupPointer` and `GroupMemberPointer` to Token2022 mints. However, there's no CLI support for adding the actual `TokenGroup` or `TokenGroupMember` extensions to mints.

#### Solution
Add the necessary commands!

**Comment by tonton-sol** [comment]
Can you expand on what is needed for this? Is this not taken care of by the `--enable-group` flag?

**Comment by buffalojoec** [comment]
@tonton-sol All of the Token Group support should be present in the Token Client, but we don't have it in the CLI yet. If you take a look at Token Metadata for reference, we're basically just setting up clap commands for driving the Token Client and using those functions!

https://github.com/solana-labs/solana-program-library/blob/23916b24700cd222ad2397e9de8162ad6980e2e8/token/cli/src/command.rs#L514

The CLI only has the ability to create Group and Group Member pointers. You can pass an address or use `--enable-group` or `--enable-member` to set the pointer to the mint itself. You can't run `InitializeGroup` or the others from the interface.

https://github.com/solana-labs/solana-program-library/commit/ad86abe10f83f180fba0257451450182c875d04a

**Comment by tonton-sol** [comment]
Understood, I didn't catch that. I'll take a look at it!

**Comment by tonton-sol** [comment]
A few questions: 

1. `spl-token initialize-group` should only invoke  `token_group_initialize_with_rent_transfer()`? and requires the mint and size?
2. `spl-token initialize-member` should only invoke  `token_group_initialize_with_rent_transfer()` and requites the mint and group-mint?
3.  Should I also add the cli functionality for updating? Or should that be in a separate PR?

**Comment by tonton-sol** [comment]
Also, I want to make sure I understand the spirit of the extensions based on reading the tests. You can create a group account then initialize it with a size. That size is the max number of members, say 10. You can then create a member, then initialize it with the group-mint. This gives 1 member. You can then use the member pointer extension, point to the first member mint and create the remaining 9. If you want to create another separate but related group of 10 members, you would use a group pointer. Then you would need to create and initialize a new member mint and initialize it, and repeat the process for minting the member pointers. Is this correct?

**Comment by buffalojoec** [comment]
> Also, I want to make sure I understand the spirit of the extensions based on reading the tests. You can create a group account then initialize it with a size. That size is the max number of members, say 10.

Correct! Max size is maximum number of members.

>You can then create a member, then initialize it with the group-mint. This gives 1 member. You can then use the member pointer extension, point to the first member mint and create the remaining 9.

You've got this one backwards. You first add `MemberPointer` when you create the mint for the group member. After the mint is created, you can `initialize_group_member` with both the group and member mints, thus initializing it as a member of the provided group. The group's authority is required here since the group is writable, and that current count of "number of member" (or `size`) is incremented in the group.

> If you want to create another separate but related group of 10 members, you would use a group pointer. Then you would need to create and initialize a new member mint and initialize it, and repeat the process for minting the member pointers. Is this correct?

I don't think I'm 100% clear on what you're asking here, but it sounds like you're referring to making sub-groups. If so, you'd basically have something like this:
1. Create the parent group
2. Create the child group, which is a member of the parent group
3. Create the members of the child group

In other words:

```
parent group
            ====== child member
                   child group
                                ====== grandchild member
```

In this case, you'd have the following extensions:
1. `GroupPointer` | `TokenGroup`
2. `GroupPointer`, `MemberPointer` | `TokenGroup`, `TokenGroupMember`
3. `MemberPointer` | `TokenGroupMember`

I guess this makes me realize I don't believe the client allows for 2), but the program does.

**Comment by buffalojoec** [comment]
> A few questions:
> 
> 1. `spl-token initialize-group` should only invoke  `token_group_initialize_with_rent_transfer()`? and requires the mint and size?
> 2. `spl-token initialize-member` should only invoke  `token_group_initialize_memeber_with_rent_transfer()` and requires the mint and group-mint?
> 3. Should I also add the cli functionality for updating? Or should that be in a separate PR?

Yep 1) and 2) are correct. For 3), yeah let's add one at a time per PR! So first one can be `command_initialize_group`.

**Comment by buffalojoec** [comment]
Thanks @tonton-sol !! This should be all wrapped!
